### PR TITLE
Cancel rentals fix

### LINKED ISSUE #372: cannot cancel rentals
looks like canceling rentals runs into an error.  although it seems like it deletes in reservation api.

### PR DESCRIPTION
**Pull request by tomecho**
fixes #372

### CODE REVIEW

File: app/models/rental.rb [hunk 1]
```diff
@@ -168,6 +167,12 @@ def delete_reservation(uuid)
     return false
   end
 
+  # same as delete_reservations but it aborts the destroy
+  def before_destroy_delete_reservations
+    delete_reservations # will add errors to obj
+    throw(:abort) if errors.any? # abort a #destroy
```
Comment (Anbranin) [inline app/models/rental.rb]: Would it be possible to use a [transaction](http://api.rubyonrails.org/classes/ActiveRecord/Transactions/ClassMethods.html) instead?

Comment (tomecho) [reply app/models/rental.rb]: I don't think they are interchangeable.  The throw abort is to abort a ActiveRecord destroy action.  However now that I think of it, I think rails won't even complete the destroy unless errors are empty.

**Comment by tomecho** [comment]
this is @Cobchise's problem now

File: app/models/rental.rb (hunk 1 above)
Comment (tomecho) [reply app/models/rental.rb]: jsut checked and it is necessary to explicitly say `throw(:abort)`, just having the errors won't do anything.

**Comment by Anbranin** [comment]
@Cobchise do you think this is safe to merge, or did you want to review it and then merge? Either way, can we close this at some point?

**Comment by Cobchise** [comment]
I don't see the need to merge this, I believe the problem was solved with #379. I'll chat through why on Monday

**Pull request closed without merging**